src/GitHub.Exports/Models/ILocalRepositoryModel.cs

```diff
@@ -18,22 +15,6 @@ public interface ILocalRepositoryModel : IRepositoryModel, INotifyPropertyChange
         /// <summary>
         /// Gets the current branch.
         /// </summary>
-        IBranch CurrentBranch { get; }
-
-        /// <summary>
-        /// Updates the url information based on the local path
-        /// </summary>
-        void Refresh();
-
-        /// <summary>
-        /// Generates a http(s) url to the repository in the remote server, optionally
-        /// pointing to a specific file and specific line range in it.
-        /// </summary>
-        /// <param name="linkType">The type of repository link to create</param>
-        /// <param name="path">The file to generate an url to. Optional.</param>
-        /// <param name="startLine">A specific line, or (if specifying the <paramref name="endLine"/> as well) the start of a range</param>
-        /// <param name="endLine">The end of a line range on the specified file.</param>
-        /// <returns>An UriString with the generated url, or null if the repository has no remote server configured or if it can't be found locally</returns>
-        Task<UriString> GenerateUrl(LinkType linkType, string path = null, int startLine = -1, int endLine = -1);
+        IBranch CurrentBranch { get; set; }
```

Comment:
It looks like we can't construct the `CurrentBranch` as soon as the repository change event fires. I think it might make sense to remove this property and construct the current branch model when needed.

Reply:
Creating the current branch model is now the responsibility of `IGitService` and no longer a property on the repository model.